datahub/bed_api/tests/conftest.py

```diff
@@ -0,0 +1,718 @@
+    PolicyArea,
+    Salutation,
+    SectorsAffected,
+    Sentiment,
+    TopIssuesByRank,
+    TransparencyStatus,
+)
+from datahub.bed_api.factories import BedFactory
+from datahub.bed_api.models import (
+    EditAccount,
+    EditContact,
+    EditEvent,
+    EditEventAttendee, EditPolicyIssues,
+)
+from datahub.bed_api.repositories import (
+    AccountRepository,
+    ContactRepository,
+    EventAttendeeRepository,
+    EventRepository,
+    PolicyIssuesRepository,
+)
+from datahub.bed_api.tests.test_utils import remove_newline
+from datahub.core.constants import Country, UKRegion
+
+
+@pytest.fixture
+def salesforce():
+    """Create salesforce instance using the BedFactory"""
+    bed_factory = BedFactory()
+    salesforce = bed_factory.create()
+    return salesforce
+
+
+@pytest.fixture
+def contact_repository(salesforce):
+    """
+    Creates instance of contact repository
+    :param salesforce: BedFactory creating an instance of salesforce
+    :return: Instance of ContactRepository
+    """
+    repository = ContactRepository(salesforce)
+    return repository
+
+
+@pytest.fixture
+def account_repository(salesforce):
+    """
+    Creates instance of account repository
+    :param salesforce: BedFactory creating an instance of salesforce
+    :return: Instance of AccountRepository
+    """
+    repository = AccountRepository(salesforce)
+    return repository
+
+
+@pytest.fixture
+def event_repository(salesforce):
+    """
+    Creates instance of event repository
+    :param salesforce: BedFactory creating an instance of salesforce
+    :return: Instance of EventRepository
+    """
+    repository = EventRepository(salesforce)
+    return repository
+
+
+@pytest.fixture
+def event_attendee_repository(salesforce):
+    """
+    Creates instance of event repository
+    :param salesforce: BedFactory creating an instance of salesforce
+    :return: Instance of EventRepository
+    """
+    repository = EventAttendeeRepository(salesforce)
+    return repository
+
+
+@pytest.fixture
+def policy_issues_repository(salesforce):
+    """
+    Creates instance of policy issues repository
+    :param salesforce: BedFactory creating an instance of salesforce
+    :return: Instance of EventRepository
+    """
+    repository = PolicyIssuesRepository(salesforce)
+    return repository
+
+
+@pytest.fixture
+def generate_high_level_sector(faker):
+    """
+    Generate random high level sector
+    :param faker: Faker Library
+    :return: Random high level sector value
+    """
+    high_level_sector = faker.random_element(
+        elements=(
+            HighLevelSector.defense,
+            HighLevelSector.food_and_agriculture,
+            HighLevelSector.energy,
+            HighLevelSector.financial_services,
+            HighLevelSector.environmental_services,
+            HighLevelSector.education_and_research,
+            HighLevelSector.creative_industries,
+            HighLevelSector.advanced_manufacturing,
+        ),
```

Comment:
```suggestion
        elements=HighLevelSector._member_map_.values()
```
I'm presenting this more as an option than a explicit suggestion. Part of me likes the explicit nature of these options, but I worry that tests won't get updated as more values are added to the enumeration. As far as I'm concerned this decision isn't blocking the PR, I think it's perfectly fine for this to stay as-is.


A related point is that given that these tests are probabilistic, it may well be that adding a value to the enumeration won't fail a test at that point for the PR adding that enumeration value, the CI test execution probably won't be pick the new value. I think there's a separate conversation to have about whether to have these tests run as part of a `pytest.mark.parameterize'; depending on how long their execution takes. 

This same comment also goes for the rest of the fixtures below where every value in the enumeration is passed into `faker.random_element`.

Reply:
That is so much better and what I wanted to do in the first place ... will do